pkg/apic/specsecuritybuilder.go

```diff
@@ -0,0 +1,370 @@
+package apic
+
+import (
+	"fmt"
+
+	"github.com/getkin/kin-openapi/openapi2"
+	"github.com/getkin/kin-openapi/openapi3"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+// used by oas spec parsers to start the builder
+func newSpecSecurityBuilder(oasMajorVersion int) SecurityBuilder {
+	return &specSecurity{
+		oasMajorVersion: oasMajorVersion,
+	}
+}
+
+// first select the type of security we are building
+type SecurityBuilder interface {
+	IsHTTPBasic() HTTPBasicSecurityBuilder
```

Comment:
Is sounds like a check, may be For... or may be just HTTPBasic() to get the builder of specific type